src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -452,18 +451,9 @@ else if (first.equals(cur)) {
             }
         });
 
-        try {
-            this.flush(this.options.getConnectionTimeout());
-        }
-        catch (Exception exp) {
-            this.processException(exp);
-        }

```

Comment:
I believe that the original intention here was to know when the re-subscribe messages, queued just before this block, finished publishing, meaning the reconnect mode had completed and the reconnect mode could be cleared. The problem was that with very large number of subscriptions, the flush could fail, triggering a reconnect loop.
This has been replaced with a marker in the reconnect queue, set via `writer.enterWaitingForEndReconnectMode()` which the writer will then automatically switch the mode itself when it reaches the marker.